packages/batch-submitter/src/transaction-chain-contract.ts

```diff
@@ -38,17 +62,21 @@ const appendSequencerBatch = async (
   batch: AppendSequencerBatchParams,
   options?: TransactionRequest
 ): Promise<TransactionResponse> => {
-  const methodId = keccak256(
-    Buffer.from(APPEND_SEQUENCER_BATCH_METHOD_ID)
-  ).slice(2, 10)
-  const calldata = encodeAppendSequencerBatch(batch)
   return OVM_CanonicalTransactionChain.signer.sendTransaction({
     to: OVM_CanonicalTransactionChain.address,
-    data: '0x' + methodId + calldata,
+    data: getEncodedCalldata(batch),
     ...options,
   })
 }
 
```

Comment:
String manipulation is so meh due to needing to slice off `0x` all the time. It makes things bug prone - its not obvious to me that `encodeAppendSequencerBatch` doesn't return a string with a `0x` prefix just by looking here

Reply:
Ok I added `remove0x` to methodId & calldata. I'm confident that it doesn't return `0x` because it was working before, but I agree it's probably better practice to be safe & sanitize the input wherever possible.